### PR TITLE
PUBDEV-7371: Add an option to specify multiple Hadoop config files

### PR DESCRIPTION
**Pull request by michalkurka**
example:

    java -jar h2o.jar -hdfs_config core-site.xml -hdfs_config hdfs-site.xml

### CODE REVIEW

**Comment by michalkurka** [comment]
@mn-mikke does SW need to reflect this change?

File: h2o-core/src/main/java/water/H2O.java [hunk 1]
```diff
@@ -591,7 +591,7 @@ else if (s.matches("nthreads")) {
       }
       else if (s.matches("hdfs_config")) {
         i = s.incrementAndCheck(i, args);
-        trgt.hdfs_config = args[i];
+        trgt.hdfs_config = ArrayUtils.append(trgt.hdfs_config, args[i]);
```
Comment (Pscheidl) [inline h2o-core/src/main/java/water/H2O.java]: Enlarging the array here is fine, as we don't know the number of HDFS confings in advance.

File: h2o-persist-hdfs/src/main/java/water/persist/PersistHdfs.java [hunk 2]
```diff
@@ -48,17 +48,17 @@
   };
 
   // Global HDFS initialization
-  // FIXME: do not share it via classes, but initialize it by object
```
Comment (Pscheidl) [inline h2o-persist-hdfs/src/main/java/water/persist/PersistHdfs.java]: :+1:

File: h2o-persist-hdfs/src/main/java/water/persist/PersistHdfs.java [hunk 3]
```diff
@@ -48,17 +48,17 @@
...
-    if( H2O.ARGS.hdfs_config != null ) {
-      conf = new Configuration();
-      File p = new File(H2O.ARGS.hdfs_config);
-      if( !p.exists() ) H2O.die("Unable to open hdfs configuration file " + p.getAbsolutePath());
-      conf.addResource(new Path(p.getAbsolutePath()));
-      Log.debug("resource ", p.getAbsolutePath(), " added to the hadoop configuration");
+    final Configuration conf = new Configuration();
+    if (H2O.ARGS.hdfs_config != null && H2O.ARGS.hdfs_config.length > 0) {
+      for (String config : H2O.ARGS.hdfs_config) {
+        File p = new File(config);
```
Comment (Pscheidl) [inline h2o-persist-hdfs/src/main/java/water/persist/PersistHdfs.java]:
```suggestion
        final File p = new File(config);
```

Tiny detail, as we started to use final 3 lines above.

File: h2o-persist-hdfs/src/main/java/water/persist/PersistHdfs.java [hunk 4]
```diff
@@ -48,17 +48,17 @@
...
+        if (!p.exists())
```
Comment (Pscheidl) [inline h2o-persist-hdfs/src/main/java/water/persist/PersistHdfs.java]:
```suggestion
        if (!p.exists()) {
          H2O.die("Unable to open hdfs configuration file " + p.getAbsolutePath());
        }
```

File: h2o-persist-hdfs/src/main/java/water/persist/PersistHdfs.java [hunk 5]
```diff
@@ -48,17 +48,17 @@
...
+          H2O.die("Unable to open hdfs configuration file " + p.getAbsolutePath());
```
Comment (Pscheidl) [inline h2o-persist-hdfs/src/main/java/water/persist/PersistHdfs.java]:
```suggestion
        if (!p.exists()) {
          H2O.die("Unable to open hdfs configuration file " + p.getAbsolutePath());
        }
```

Just a matter of style. Don't worry about it if you don't like it. This way, the boundary of the block is much more clearly defined.

**Comment by mn-mikke** [comment]
@michalkurka IMHO, this shouldn't break anything in SW. The question is whether we should expose ability to set multiple files manually as well. https://github.com/h2oai/sparkling-water/blob/2b4dbeb5ade89605f4616c31f71bcc263069ca5c/core/src/main/scala/ai/h2o/sparkling/backend/internal/InternalBackendConf.scala#L65, but this is a rare case.

The feature that @pavel implemented for the internal backend dumps the whole spark hadoop configuration into one big file and passes that.

File: h2o-persist-hdfs/src/main/java/water/persist/PersistHdfs.java [hunk 6]
```diff
@@ -48,17 +48,17 @@
...
+        conf.addResource(new Path(p.getAbsolutePath()));
+        Log.debug("resource ", p.getAbsolutePath(), " added to the hadoop configuration");
+      }
     } else {
-      conf = new Configuration();
```
Comment (Pscheidl) [inline h2o-persist-hdfs/src/main/java/water/persist/PersistHdfs.java]: :+1:

**Comment by Pscheidl** [comment]
> @michalkurka IMHO, this shouldn't break anything in SW. The question is whether we should expose ability to set multiple files manually as well. https://github.com/h2oai/sparkling-water/blob/2b4dbeb5ade89605f4616c31f71bcc263069ca5c/core/src/main/scala/ai/h2o/sparkling/backend/internal/InternalBackendConf.scala#L65, but this is a rare case.
> 
> The feature that @pavel implemented for the internal backend dumps the whole spark hadoop configuration into one big file and passes that.

I can only agree. 100% true.

**Pull request merged**